lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -804,16 +820,37 @@ sub fix_tests{
     my $locale = $request->{_locale};
 
     my $table = $request->{dbh}->quote_identifier($request->{table});
-    my $edit = $request->{dbh}->quote_identifier($request->{edit});
     my $where = $request->{id_where};
-    my $sth = $request->{dbh}->prepare(
-            "UPDATE $table SET $edit = ? where $where = ?"
-    );
+
+    my @edits;
+    my $i = 1;
+    # Move around Can't use string "ARRAY as an ARRAY ref while "strict refs" in use
+    while (defined $request->{"edit_$i"}) {
+      push @edits, $request->{"edit_$i"};
+      $i++;
+    }
+    my $sth = $request->{insert}
+      ? $request->{dbh}->prepare(
+            "INSERT INTO $table(" . join(',',@edits) . ") VALUES(" .
+                join(',',map { '?' } @edits) . ")" )
+      : $request->{dbh}->prepare(
+            "UPDATE $table SET " .
+                join(',',map {$request->{dbh}->quote_identifier($_) . " = ?"} @edits) .
+                " where $where = ?");
```

Comment:
A better solution would be to use the value from the Upgrade_Tests array element to which this data belongs